.drone.yml

```diff
@@ -1,24 +1,55 @@
 ---
 kind: pipeline
+type: docker
 name: testing
 
 platform:
   os: linux
   arch: amd64
 
 steps:
+- name: environment
+  pull: always
```

Comment:
we should avoid pull always because this will force docker to pull a new image on every execution, which will slow down the pipeline execution.

Reply:
Now I'm wondering if my thinking is totally wrong.

When I have a tag like `golang:1.15` because it can change I put it as `pull: always` since today `golang:1.15.2` is the latest release but tomorrow it could get bumped to `golang:1.15.3` so if there's no `pull: always` it'd end up with `golang:1.15.2` perpetually.

Reply:
The only difference between `pull: if-not-exists` and `pull: never` is whether or not the runner should pull the image and retry if `docker create` fails due to a missing image.  There is no real benefit to `pull: never` in this case, especially considering you leave yourself susceptible to a race condition if the image is purged from the local cache after the initial pull.

In terms of your thinking, you are correct that `pull: always` would ensure you always have the latest minor version of Drone, but this comes at a cost of pulling the full image on every pipeline execution. This probably comes down to personal preference, but my belief is the cost of constantly downloading images outweighs the benefits of minor Go updates given the stability of the Go language and runtime.

Reply:
Alrighty well the `pull`s are removed across the board